File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l=[]
        for i in range(len(nums)):
            if target-nums[i] in l:
                return [i,l.index(target-nums[i])]

            l.append(nums[i])

    
        
# @lc code=end

File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start

import java.util.HashMap;

class Solution {
    HashMap<Integer,Integer> mp=new HashMap<>();
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            if(this.mp.containsKey(target-nums[i]))
            {
                return new int[]{this.mp.get(target-nums[i]),i};
            }
            this.mp.put(nums[i], i);
        }
        return new int[]{-1,-1};
    }
}
// @lc code=end

